design/helm.md

```diff
@@ -0,0 +1,145 @@
+# Helm Addons
+
+To be an addon in a Kurl cluster, they must be contributed into the repo and maintained across new breaking changes and new versions.
+This proposal will give users the flexibility to specify their own addons through a helm chart support in the Kurl installer spec.
+
+
+## Goals
+- Allow any Helm chart to be specified as part of a Kurl spec.
+
+## Non Goals
+
+- Verify authenticity, trustworthyness and "goodness" of helm addons.
+- Validate Helm values.
+
+## Background
+
+Users of Kurl have a limited set of available cluster addons and often want to include ones that are unsupported.
+There is friction in the [kurl development process](https://github.com/replicatedhq/kURL/blob/master/test/dev.md), [contributing new addons](https://kurl.sh/docs/add-on-author/) and addon maintanence. 
+
+Helm is a popular tool for managing deployment configuration that allows application/addon developers or the community to maintain a consistent interface for deployment through a chart.
+By allowing Kurl users to specifify a chart and its values, we can allow customization of kutl with new addons while delegating addons support and maintance to its community.
+
+## High-Level Design
+
+Kurl will bundle the Helm binary into the common tools bundle already delivering krew and kubectl plugins. The install.sh script will perform the following in the online case:
+1. Unarchive Helm and move into the `$PATH` and `/usr/bin`
+1. For each release, check if the release exists. 
+1. If the release exists, perform a `helm upgrade`.
+1. If the release doesn't exist, perform a `helm install`
+
+For airgapped the chart and artifacts need to be packaged into a bundle. The builder should perform the following:
+1. `helm pull` the repo
+1. `helm template` to render the chart with the provided values
+1. Generate an adhoc `manifest` file by parsing `image` specs from the rendered yaml.
```

Comment:
what if it includes an operator? is there a common way to specify images in a helm chart? if not could you add an additionalImages property for airgap purposes?

Reply:
That's a really god question. AFAIK there isn't a way to specify this kind of relationship in helm since the operator doesn't need to declare it in a manifest or as a dependency.  `additionalImages` would be a good way to deal with this. I'll add an example.